src/integrations/rapier/fluids_pipeline.rs

```diff
@@ -158,102 +158,104 @@ impl<'a> CouplingManager for ColliderCouplingManager<'a> {
                 boundaries.get_mut(coupling.boundary),
             ) {
                 // Update the boundary's ability to receive forces.
-                let body = self.bodies.get(collider.parent());
-                if let Some(body) = body {
-                    if !body.is_dynamic() {
-                        boundary.forces = None;
-                    } else {
-                        boundary.forces = Some(RwLock::new(Vec::new()));
-                        boundary.clear_forces(true);
+                if let Some(parent) = collider.parent() {
+                    let body = self.bodies.get(parent);
+                    if let Some(body) = body {
```

Comment:
Let's merge both `if` into a single one so we don't have to add one level of indentation (and this will reduce the size of this diff):
```suggestion
                if let Some(body) = collider.parent().map(|p| self.bodies.get(p)) {
```

Reply:
Done